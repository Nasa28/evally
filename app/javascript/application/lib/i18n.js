const messages = {
  en: {
    buttons: {
      archive: 'Archive',
      cancel: 'Cancel',
      complete: 'Complete',
      copy: 'Copy',
      create: 'Create',
      delete: 'Delete',
      edit: 'Edit',
      hide: 'Hide',
      show: 'Show'
    },
    destroy_modals: {
      default: {
        title: 'Delete',
        confirm: 'Are you sure you want to delete it?'
      },
      evaluation: {
        title: 'Delete evaluation',
        confirm: 'Are you sure you want to delete this evaluation?'
      },
      employee: {
        title: 'Delete employee',
        confirm: 'Are you sure you want to delete this employee?'
      },
      template: {
        title: 'Delete template',
        confirm: 'Are you sure you want to delete this template?'
      }
    },
    employees: {
      title: 'Employees',
      buttons: {
        add_new: 'New employee',
        permalink: 'Permalink',
        edit: 'Edit employee',
        archive: 'Archive employee',
        delete: 'Delete employee'
      },
      flashes: {
        archive: {
          success: 'Employee has been succefully archived.',
          error: 'Employee cannot be archived due to some error: {reason}'
        },
        create: {
          success: 'Employee has been succefully created.',
          error: 'Employee cannot be created due to some error: {reason}'
        },
        edit: {
          success: 'Employee has been succefully updated.',
          error: 'Employee cannot be updated due to some error: {reason}'
        },
        fetch: {
          error: 'Employees cannot be loaded due to some error: {reason}'
        },
        delete: {
          success: 'Evaluation has been succefully deleted.',
          error: 'Evaluation cannot be destroyed due to some error: {reason}'
        }
      },
      forms: {
        archive: {
          title: 'Archive employee',
          warning: 'This action cannot be undo. All drafted evaluations of the employee will be destroyed!',
          text: 'Please, provide the reason to archive the employee: ',
          too_experienced: 'Too experienced',
          released: 'Released',
          released_at: 'Released at'
        },
        permalink: {
          title: 'Share permalink',
          text: 'Copy the permalink below to share the latest evaluation:'
        },
        create_title: 'Create employee',
        edit_title: 'Edit employee',
        first_name: 'First name',
        last_name: 'Last name',
        position: 'Position',
        hired_at: 'On board since',
        next_review: 'Next review'
      },
      messages: {
        select: 'Select employee to see his latest evaluation.',
        not_evaluated: '{name} has not been evaluated yet.'
      },
      sidebar: {
        search: 'Search ...',
        subheader: 'List of employees',
        no_items: 'There are no employees'
      }
    },
    evaluations: {
      title: 'Evaluations',
      buttons: {
        add_new: 'New evaluation',
        complete: 'Complete evaluation',
        save_draft: 'Save draft',
        reset: 'Undo changes',
        delete: 'Delete evaluation'
      },
      flashes: {
        create: {
          success: 'Evaluation has been succefully created.',
          error: 'Evaluation cannot be created due to some error: {reason}'
        },
        fetch: {
          error: 'Evaluation cannot be loaded due to some error: {reason}'
        },
        save: {
          success: 'Evaluation has been succefully updated.',
          error: 'Evaluation cannot be updated due to some error: {reason}'
        },
        complete: {
          success: 'Evaluation has been succefully completed.',
          error: 'Evaluation cannot be completed due to some error: {reason}'
        },
        delete: {
          success: 'Evaluation has been succefully deleted.',
          error: 'Evaluation cannot be destroyed due to some error: {reason}'
        }
      },
      forms: {
        complete: {
          title: 'Complete evaluation',
          question: 'Are you sure you want to complete this evaluation?',
          warning: 'This action cannot be undo.',
          next_label: 'Next evaluation at'
        },
        create_title: 'Create draft',
        select_employee: 'Select employee',
        select_template: 'Select template',
        employee_label: 'Employee',
        template_label: 'Template',
        bool_no: 'No',
        bool_yes: 'Yes',
        hired_at: 'hired at',
        next_review: 'next review',
        completed_at: 'completed at'
      },
      message: 'Select draft evaluation or create a new one.',
      sidebar: {
        search: 'Search ...',
        subheader: 'Evaluations drafts',
        subtitle: 'based on \'{name}\' template',
        no_items: 'There are no evaluations drafts'
      }
    },
    dashboard: {
      title: 'Hello {name}!',
      common: {
        as: ' as ',
        first_time: 'First time'
      },
      drafts: {
        title: 'Drafts',
        edit_tooltip: 'Edit',
        updated: 'Updated {date}'
      },
      upcoming: {
        title: 'Upcoming',
        new_tooltip: 'Create'
      },
      activities: {
        title: 'Activities',
        no_activities: 'There are no activities today',
        see_more: 'See more'
      }
    },
    navbar: {
      dashboard: 'Dashboard',
      evaluations: 'Evaluations',
      employees: 'Employees',
      templates: 'Templates',
      archive: 'Archive',
      settings: 'Settings',
      logout: 'Logout'
    }
  },

  pl: {
    buttons: {
      archive: 'Archiwizuj',
      cancel: 'Anuluj',
      complete: 'Zakończ',
      copy: 'Skopiuj',
      create: 'Utwórz',
      delete: 'Usuń',
      edit: 'Edytuj',
      hide: 'Ukryj',
      show: 'Pokaż'
    },
    destroy_modals: {
      default: {
        title: 'Usuń',
        confirm: 'Czy na pewno chcesz to usunąć?'
      },
      evaluation: {
        title: 'Usuń ocenę',
        confirm: 'Czy na pewno chcesz usunąć ocenę tego pracownika?'
      },
      employee: {
        title: 'Usuń pracownika',
        confirm: 'Czy na pewno chcesz usunąć tego pracownika?'
      },
      template: {
        title: 'Usuń szablon',
        confirm: 'Czy na pewno chcesz usunąć ten szablon?'
      }
    },
    employees: {
      title: 'Pracownicy',
      buttons: {
        add_new: 'Nowy pracownik',
        permalink: 'Permalink',
        edit: 'Edytuj pracownika',
        archive: 'Przenieś do archiwum',
        delete: 'Usuń pracownika'
      },
      forms: {
        archive: {
          title: 'Przenieś do archiwum',
          warning: 'Nie ma możliwości cofnięcia tej akcji. Wszystkie szkice ocen tego pracownika zostaną usunięte!',
          text: 'Proszę podać powód przeniesienia pracownika do archiwum: ',
          too_experienced: 'Bogate doświadczenie',
          released: 'Koniec współpracy',
          released_at: 'Współpraca zakończona'
        },
        permalink: {
          title: 'Udostępnij link',
          text: 'Skopiuj link z dostępem do aktualnej oceny tego pracownika:'
        },
        create_title: 'Utwórz pracownika',
        edit_title: 'Edytuj pracownika',
        first_name: 'Imię',
        last_name: 'Nazwisko',
        position: 'Stanowisko',
        hired_at: 'Zatrudniony od',
        next_review: 'Następne review'
      },
      flashes: {
        archive: {
          success: 'Pracownik został pomyślnie zarchiwizowany.',
          error: 'Wystąpił błąd podczas archiwizowania pracownika: {reason}'
        },
        create: {
          success: 'Pracownik został pomyślnie dodany.',
          error: 'Wystąpił błąd podczas dodawania pracownika: {reason}'
        },
        edit: {
          success: 'Pracownik został pomyślnie zapisany.',
          error: 'Wystąpił błąd podczas zapisywania pracownika: {reason}'
        },
        fetch: {
          error: 'Wystąpił błąd podczas wczytywania pracowników: {reason}'
        },
        delete: {
          success: 'Pracownik został pomyślnie usunięty.',
          error: 'Wystąpił błąd podczas usuwania pracownika: {reason}'
        }
      },
      messages: {
        select: 'Wybierz pracownika z listy, aby zobaczyć jego aktualną ocenę.',
        not_evaluated: '{name} nie został jeszcze oceniony.'
      },
      sidebar: {
        search: 'Szukaj ...',
        subheader: 'Lista pracowników',
        no_items: 'Brak pracowników'
      }
    },
    evaluations: {
      title: 'Oceny',
      buttons: {
        add_new: 'Nowa ocena',
        complete: 'Zakończ ocenę',
        save_draft: 'Zapisz szkic',
        reset: 'Cofnij zmiany',
        delete: 'Usuń ocenę'
      },
      flashes: {
        create: {
          success: 'Ocena została pomyślnie utworzona.',
          error: 'Wystąpił błąd podczas tworzenia szkicu: {reason}'
        },
        fetch: {
          error: 'Wystąpił błąd podczas wczytywania ocen: {reason}'
        },
        save: {
          success: 'Ocena została pomyślnie zapisana.',
          error: 'Wystąpił błąd podczas zapisu: {reason}'
        },
        complete: {
          success: 'Ocena została pomyślnie zakończona.',
          error: 'Wystąpił błąd podczas zapisu: {reason}'
        },
        delete: {
          success: 'Ocena została pomyślnie usunięta.',
          error: 'Wystąpił błąd podczas usuwania: {reason}'
        }
      },
      forms: {
        complete: {
          title: 'Zakończ ocenę',
          question: 'Czy na pewno chcesz zakończyć ocenę pracownika?',
          warning: 'Ta akcja jest ostateczna i nie można jej cofnąć.',
          next_label: 'Następne review'
        },
        create_title: 'Utwórz szkic',
        select_employee: 'Wybierz pracownika',
        select_template: 'Wybierz szablon',
        employee_label: 'Pracownik',
        template_label: 'Szablon',
        bool_no: 'Nie',
        bool_yes: 'Tak',
        hired_at: 'zatrudniono',
        next_review: 'następne review',
        completed_at: 'zakończono'
      },
      message: 'Wybierz szkic oceny z listy lub stwórz nowy.',
      sidebar: {
        search: 'Szukaj ...',
        subheader: 'Szkice ocen',
        subtitle: 'według szablonu \'{name}\'',
        no_items: 'Brak szkiców ocen'
      }
    },
    dashboard: {
      title: 'Cześć {name}!',
      common: {
        as: ' jako ',
        first_time: 'Pierwszy raz'
      },
      drafts: {
        title: 'Szkice',
        edit_tooltip: 'Edytuj',
        updated: 'Edytowano {date}'
      },
      upcoming: {
        title: 'Nadchodzące',
        new_tooltip: 'Utwórz'
      },
      activities: {
        title: 'Aktywności',
        no_activities: 'Brak aktywności',
        see_more: 'Więcej'
      }
    },
    navbar: {
      dashboard: 'Kokpit',
      evaluations: 'Oceny',
      employees: 'Pracownicy',
      templates: 'Szablony',
      archive: 'Archiwum',
      settings: 'Ustawienia',
      logout: 'Wyloguj'
    }
  }
}

export default messages