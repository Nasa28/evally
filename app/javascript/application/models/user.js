import { Model, Collection } from 'vue-mc'


/**
 * User model
 */
class User extends Model {

  defaults() {
    return {
      email: '',
      first_name: '',
      last_name: ''
    }
  }

  fullname() {
    return [this.first_name, this.last_name].join(' ')
  }

  initials() {
    return [this.first_name[0], this.last_name[0]].join('').toUpperCase()
  }
}

export { User }
