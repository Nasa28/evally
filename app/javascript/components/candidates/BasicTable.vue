<template>
  <div class="box">
    <v-layout row wrap>
      <v-flex xs12 lg6>
        <v-text-field
          v-model="search"
          append-icon="mdi-magnify"
          :label="$t('components.employees.table.search')"
          filled
        />
      </v-flex>

      <v-flex xs12>
        <v-data-table
          :headers="headers"
          :items="candidates.models"
          :search="search"
          :loading="loading"
        >
          <template #item.action="{ item }">
            <v-tooltip v-if="editable" bottom>
              <template #activator="{ on }">
                <v-icon
                  @click="$emit('edit', item.id)"
                  v-on="on"
                  class="mx-2"
                  small
                >
                  mdi-pencil
                </v-icon>
              </template>

              <span>{{ $t('shared.tooltips.edit') }}</span>
            </v-tooltip>
          </template>

          <template #item.fullname="{ item }">
            <router-link :to="{ name: 'candidate_path', params: { id: item.id }}">
              {{ item.fullname }}
            </router-link>
          </template>

        </v-data-table>
      </v-flex>
    </v-layout>
  </div>
</template>

<script>
import { CandidatesList } from '@models/candidate'

export default {
  name: 'BasicTable',
  props: {
    loading: {
      type: Boolean,
      required: true,
      default: false
    },
    candidates: {
      type: CandidatesList,
      required: true,
      default: () => new CandidatesList()
    },
    editable: {
      type: Boolean,
      required: true,
      default: false
    }
  },
  data() {
    return {
      search: '',
      headers: [
        {
          sortable: false,
          value: 'action',
        },
        {
          text: this.$t('components.employees.table.cols.name'),
          value: 'fullname'
        }
      ]
    }
  }
}
</script>
