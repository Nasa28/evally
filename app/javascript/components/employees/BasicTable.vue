<template>
  <v-layout row wrap>
    <v-flex xs4>
      <v-text-field
        v-model="search"
        append-icon="search"
        :label="$t('employees.table.search')"
      />
    </v-flex>

    <v-flex xs12>
      <div class="employees__table">
        <v-data-table
          :headers="headers"
          :items="employees.models"
          :search="search"
          :rows-per-page-items="perPageItems"
          :loading="loading"
        >
          <template #items="props">
            <td :style="{ padding: '0 5px' }">
              <v-tooltip right>
                <v-btn
                  @click="$emit('edit', props.item.id)"
                  slot="activator"
                  icon
                  flat
                >
                  <v-icon size="15">edit</v-icon>
                </v-btn>
                <span>Edit</span>
              </v-tooltip>
            </td>
            <td>
              <router-link :to="{ name: 'employee_path', params: { id: props.item.id }}">
                {{ props.item.fullname }}
              </router-link>
            </td>
            <td class="text-xs-center">
              <v-tooltip left>
                <template #activator="{ on }">
                  <span v-on="on">{{ props.item.hiredAt }}</span>
                </template>
                <span>{{ props.item.employmentTime }}</span>
              </v-tooltip>
            </td>
            <td class="text-xs-center">{{ props.item.group }}</td>
            <td class="text-xs-center">{{ props.item.position }}</td>
            <td class="text-xs-center">{{ props.item.positionSetAt }}</td>
            <td class="text-xs-center">{{ props.item.latestEvaluationDate }}</td>
          </template>
        </v-data-table>
      </div>
    </v-flex>
  </v-layout>
</template>

<script>
import { EmployeesList } from '@models/employee'

export default {
  name: 'BasicTable',
  props: {
    loading: {
      type: Boolean,
      required: true,
      default: false
    },
    employees: {
      type: EmployeesList,
      required: true,
      default: () => new EmployeesList()
    }
  },
  data() {
    return {
      search: '',
      headers: [
        {
          sortable: false,
          value: null,
        },
        {
          text: this.$t('employees.table.headers.employee_name'),
          value: 'fullname'
        },
        {
          text: this.$t('employees.table.headers.hired_at'),
          value: 'hired_at',
          align: 'center'
        },
        {
          text: this.$t('employees.table.headers.team'),
          value: 'group',
          align: 'center'
        },
        {
          text: this.$t('employees.table.headers.position'),
          value: 'position',
          align: 'center'
        },
        {
          text: this.$t('employees.table.headers.position_set_at'),
          value: 'position_set_at',
          align: 'center'
        },
        {
          text: this.$t('employees.table.headers.last_evaluation'),
          value: 'latest_evaluation_date',
          align: 'center'
        }
      ],
      perPageItems: [10, 30, { text: '$vuetify.dataIterator.rowsPerPageAll' , value: -1 }]
    }
  }
}
</script>
