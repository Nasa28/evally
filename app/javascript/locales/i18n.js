import Vue from 'vue'
import VueI18n from 'vue-i18n'

Vue.use(VueI18n)

import en from '@locales/en'
import pl from '@locales/pl'

export default new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages: {
    en,
    pl
  }
})
