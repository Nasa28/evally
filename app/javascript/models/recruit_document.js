import { Model, List } from './base'
import { getURLParams } from '@utils/helpers'

import moment from 'moment'

class RecruitDocument extends Model {
  get defaults() {
    return {
      id: null,
      gender: '',
      first_name: '',
      last_name: '',
      email: '',
      group: '',
      phone: '',
      position: '',
      source: '',
      received_at: moment().format('YYYY-MM-DD'),
      accept_current_processing: false,
      accept_future_processing: false,
      public_recruit_id: '',
      status: {
        value: '',
        label: '',
        color: '',
        required_fields: []
      },
      task_sent_at: '',
      call_scheduled_at: '',
      interview_scheduled_at: '',
      decision_made_at: '',
      recruit_accepted_at: '',
      rejection_reason: ''
    };
  }

  get fullname() {
    return [this.first_name, this.last_name].join(' ')
  }

  get receivedAt() {
    return moment(this.received_at).format('YYYY-MM-DD HH:mm')
  }

  get taskSentAt() {
    if (!this.task_sent_at) return '---'

    return moment(this.task_sent_at).format('YYYY-MM-DD HH:mm')
  }

  get callScheduledAt() {
    if (!this.call_scheduled_at) return '---'

    return moment(this.call_scheduled_at).format('YYYY-MM-DD HH:mm')
  }

  get interviewScheduledAt() {
    if (!this.interview_scheduled_at) return '---'

    return moment(this.interview_scheduled_at).format('YYYY-MM-DD HH:mm')
  }

  get decisionMadeAt() {
    if (!this.decision_made_at) return '---'

    return moment(this.task_sent_at).format('YYYY-MM-DD HH:mm')
  }

  get recruitAcceptedAt() {
    if (!this.recruit_accepted_at) return '---'

    return moment(this.recruit_accepted_at).format('YYYY-MM-DD HH:mm')
  }

  static get routes() {
    return {
      recruitDocumentsPath: '/v2/recruit_documents',
      recruitDocumentsFilterPath: payload => `/v2/recruit_documents?${getURLParams(payload)}`,
      recruitDocumentsFormPath: '/v2/recruit_documents/form',
      recruitDocumentPath: id => `/v2/recruit_documents/${id}`,
      recruitPath: publicRecruitId => `/v2/recruits/${publicRecruitId}`
    }
  }
}

class RecruitDocumentsList extends List {
  get model() {
    return RecruitDocument;
  }
}

export { RecruitDocument, RecruitDocumentsList };
