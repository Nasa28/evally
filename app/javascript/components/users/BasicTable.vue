<template>
  <div class="box">
    <v-layout row wrap>
      <v-flex xs12 lg6>
        <v-text-field
          v-model="search"
          append-icon="mdi-magnify"
          label="Search ..."
          filled
        />
      </v-flex>

      <v-flex xs12>
        <v-data-table
          :headers="headers"
          :items="users.models"
          :search="search"
          :loading="loading"
        >
          <template #item.action="{ item }">
            <v-tooltip bottom>
              <template #activator="{ on }">
                <v-icon
                  @click="$emit('edit', item.id)"
                  v-on="on"
                  class="mx-2"
                  small
                >
                  mdi-pencil
                </v-icon>
              </template>

              <span>{{ $t('shared.tooltips.edit') }}</span>
            </v-tooltip>
          </template>

          <template #item.fullname="{ item }">
            {{ item.fullname }}

            <span v-if="item.id === user.id">
              (you)
            </span>
          </template>

          <template #item.last_sign_in_at="{ item }">
            {{ item.lastSignInAt }}
          </template>
        </v-data-table>
      </v-flex>
    </v-layout>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'

import { UsersList } from '@models/user'

export default {
  name: 'BasicTable',
  props: {
    loading: {
      type: Boolean,
      required: true,
      default: false
    },
    users: {
      type: UsersList,
      required: true,
      default: () => new UsersList()
    }
  },
  data() {
    return {
      search: '',
      headers: [
        {
          sortable: false,
          value: 'action',
        },
        {
          text: 'Name',
          value: 'fullname'
        },
        {
          text: 'Email',
          value: 'email',
          align: 'center'
        },
        {
          text: 'Role',
          value: 'role',
          align: 'center'
        },
        {
          text: 'Latest Activity',
          value: 'last_sign_in_at',
          align: 'center'
        },
        {
          text: 'Status',
          value: 'status',
          align: 'center'
        },
        {
          text: 'Invitation Status',
          value: 'invitation_status',
          align: 'center'
        }
      ]
    }
  },
  computed: {
    ...mapGetters({
      user: 'SessionModule/user'
    })
  }
}
</script>
