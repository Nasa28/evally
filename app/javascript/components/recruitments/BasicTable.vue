<template>
  <div class="box">
    <server-side-table
      :composerClass="composerClass"
      :headers="headers"
      :items="recruitDocuments.models"
      :footer-props="{ 'items-per-page-options': [25, 50, 100] }"
      :server-items-length="totalCount"
      :loading="loading"
      @change:options="fetchRecruitDocuments"
    >
      <template #search="{ options, searchBy }">
        <v-flex xs12 lg3>
          <v-text-field
            :value="options.search"
            @input="searchBy"
            append-icon="mdi-magnify"
            :label="$t('components.recruitments.table.search')"
            solo
          />
        </v-flex>
      </template>

      <template #filters="{ options }">
        <v-flex xs12 lg3>
          <v-select
            v-model="options.group"
            :items="groups"
            :label="$t('components.recruitments.table.groupFilter')"
            clearable
            chips
            solo
          />
        </v-flex>

        <v-flex xs12 lg3>
          <status-select
            v-model="options.status"
            :items="statuses"
            :label="$t('components.recruitments.table.statusFilter')"
            clearable
            solo
          />
        </v-flex>

        <v-flex xs12 lg3>
          <v-select
            v-model="options.evaluatorToken"
            :items="evaluators.models"
            :label="$t('components.recruitments.table.evaluatorFilter')"
            item-value="email_token"
            item-text="fullname"
            clearable
            chips
            solo
          />
        </v-flex>
      </template>

      <template #item.action="{ item }">
        <v-tooltip
          v-if="recruitDocumentPolicy.canRemove"
          bottom
        >
          <template #activator="{ on }">
            <v-icon
              @click="$emit('delete', item.id)"
              v-on="on"
              class="mx-2"
              color="red"
              small
            >
              mdi-delete-outline
            </v-icon>
          </template>

          <span>{{ $t('shared.tooltips.delete') }}</span>
        </v-tooltip>

        <v-tooltip
          v-if="recruitDocumentPolicy.canEdit"
          bottom
        >
          <template #activator="{ on }">
            <v-icon
              @click="$router.push(item.editPath)"
              v-on="on"
              class="mx-2"
              small
            >
              mdi-pencil
            </v-icon>
          </template>

          <span>{{ $t('shared.tooltips.edit') }}</span>
        </v-tooltip>
      </template>

      <template #item.last_name="{ item }">
        <router-link :to="item.showPath">
          {{ item.fullname }}
        </router-link>
      </template>

      <template #item.status="{ item }">
        <v-chip
          :color="item.status.color"
          dark
          small
          label
        >
          <strong>{{ item.status.label }}</strong>
        </v-chip>
      </template>

      <template #item.evaluator_fullname="{ item }">
        {{ item.evaluator_fullname || '---' }}
      </template>

      <template #item.received_at="{ item }">
        {{ item.receivedAt }}
      </template>

      <template #item.accept_current_processing="{ item }">
        <v-icon>{{ item.accept_current_processing ? 'mdi-check' : 'mdi-minus' }}</v-icon>
      </template>

      <template #item.accept_future_processing="{ item }">
        <v-icon>{{ item.accept_future_processing ? 'mdi-check' : 'mdi-minus' }}</v-icon>
      </template>
    </server-side-table>
  </div>
</template>

<script>
import RecruitDocumentPolicy from '@policies/recruit_document_policy'

import { RecruitDocumentsList } from '@models/recruit_document'
import { UsersList } from '@models/user'

import StatusSelect from '@components/recruitments/StatusSelect'
import ServerSideTable from '@components/shared/ServerSideTable'

import RecruitmentsTableComposer from '@utils/data_tables/recruitments_table_composer'

export default {
  name: 'BasicTable',
  components: { StatusSelect, ServerSideTable },
  props: {
    loading: {
      type: Boolean,
      required: true,
      default: true
    },
    recruitDocuments: {
      type: RecruitDocumentsList,
      required: true,
      default: () => new RecruitDocumentsList()
    },
    evaluators: {
      type: UsersList,
      required: true,
      default: () => new UsersList()
    },
    recruitDocumentPolicy: {
      type: RecruitDocumentPolicy,
      required: true,
      default: () => new RecruitDocumentPolicy()
    },
    groups: {
      type: Array,
      required: true,
      default: () => []
    },
    statuses: {
      type: Array,
      required: true,
      default: () => []
    },
    totalCount: {
      type: Number,
      required: true,
      default: 0
    }
  },
  data() {
    return {
      composerClass: RecruitmentsTableComposer,
      headers: [
        {
          sortable: false,
          value: 'action',
        },
        {
          text: this.$t('components.employees.table.cols.name'),
          value: 'last_name'
        },
        {
          text: this.$t('components.recruitments.table.cols.email'),
          value: 'email'
        },
        {
          text: this.$t('components.recruitments.table.cols.group'),
          value: 'group'
        },
        {
          text: this.$t('components.recruitments.table.cols.position'),
          value: 'position'
        },
        {
          text: this.$t('components.recruitments.table.cols.status'),
          value: 'status'
        },
        {
          text: this.$t('components.recruitments.table.cols.currentEvaluator'),
          value: 'evaluator_fullname',
          sortable: false
        },
        {
          text: this.$t('components.recruitments.table.cols.source'),
          value: 'source'
        },
        {
          text: this.$t('components.recruitments.table.cols.receivedAt'),
          value: 'received_at'
        },
        {
          text: this.$t('components.recruitments.table.cols.acceptCurrentProcessing'),
          value: 'accept_current_processing',
        },
        {
          text: this.$t('components.recruitments.table.cols.acceptFutureProcessing'),
          value: 'accept_future_processing',
        }
      ]
    }
  },
  methods: {
    fetchRecruitDocuments(query) {
      this.$store.dispatch('RecruitDocumentsModule/fetchRecruitDocuments', query)
    }
  }
}
</script>
