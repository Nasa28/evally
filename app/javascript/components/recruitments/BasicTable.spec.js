import { shallowMount } from '@vue/test-utils'
import i18n from '@locales/i18n'

import BasicTable from './BasicTable'

import RecruitDocumentPolicy from '@policies/recruit_document_policy'

import { RecruitDocumentsList } from '@models/recruit_document'
import { User } from '@models/user'

describe('BasicTable', () => {
  const admin = new User({role: 'admin'})

  const props = {
    loading: false,
    recruitDocuments: new RecruitDocumentsList(),
    groups: ['ror', 'android'],
    statuses: ['fresh', 'accepted'],
    editable: false,
    recruitDocumentPolicy: new RecruitDocumentPolicy(admin, null)
  }

  const wrapper = shallowMount(
    BasicTable,
    {
      propsData: props,
      stubs: ['v-layout', 'v-data-table', 'v-flex', 'v-text-field', 'v-select'],
      mocks: {
        $t: key => i18n.t(key)
      }
    })

  it('has a name', () => {
    expect(BasicTable.name).toEqual('BasicTable')
  })

  it('props', () => {
    expect(wrapper.props().loading).toEqual(false)
    expect(wrapper.props().recruitDocuments.models).toHaveLength(0)
  })

  it('data.search', () => {
    expect(wrapper.vm.search).toEqual('')
  })

  it('data.headers', () => {
    expect(wrapper.vm.headers).toEqual([{
      sortable: false,
      value: 'action',
    },
    {
      text: i18n.t('components.employees.table.cols.name'),
      value: 'fullname'
    },
    {
      text: i18n.t('components.recruitments.table.cols.email'),
      value: 'email'
    },
    {
      text: i18n.t('components.recruitments.table.cols.group'),
      value: 'group',
      filterable: false
    },
    {
      text: i18n.t('components.recruitments.table.cols.position'),
      value: 'position'
    },
    {
      text: i18n.t('components.recruitments.table.cols.status'),
      value: 'status',
      filterable: false
    },
    {
      text: i18n.t('components.recruitments.table.cols.source'),
      value: 'source'
    },
    {
      text: i18n.t('components.recruitments.table.cols.receivedAt'),
      value: 'received_at'
    },
    {
      text: i18n.t('components.recruitments.table.cols.acceptCurrentProcessing'),
      value: 'accept_current_processing',
    },
    {
      text: i18n.t('components.recruitments.table.cols.acceptFutureProcessing'),
      value: 'accept_future_processing',
    }])
  })

  it('matches do snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
