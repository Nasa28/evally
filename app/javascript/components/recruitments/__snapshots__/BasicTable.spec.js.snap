// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTable matches do snapshot 1`] = `
<div class="box">
  <v-layout-stub row="" wrap="">
    <v-flex-stub xs12="" lg6="">
      <v-text-field-stub append-icon="mdi-magnify" label="Search ..." filled="" value=""></v-text-field-stub>
    </v-flex-stub>
    <v-combobox-stub items=",ror,android"></v-combobox-stub>
    <v-combobox-stub items=",fresh,accepted"></v-combobox-stub>
    <v-flex-stub xs12="">
      <v-data-table-stub headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" items="" search=""></v-data-table-stub>
    </v-flex-stub>
  </v-layout-stub>
</div>
`;
